tests/test.hs

```diff
@@ -54,28 +53,6 @@ main = defaultMain $ testGroup "Tests"
       v <- atomically $ readTVar tv
       assertBool "Thread did not stop" (not v)
 
-  , testCase "state is preserved when there are no exceptions" $ do
-      tv <- atomically $ newTVar 0
-      pid <- flip evalStateT 0 $ Immortal.create $ const $ countToFive tv
-      (do
-        delay
-        v <- atomically $ readTVar tv
-        v @?= 5) `finally` Immortal.stop pid
-
-  , testCase "state is reset when there are exceptions" $ do
-      tv <- atomically $ newTVar 0
-      let
-        computation = do
-          countToFive tv
-          liftIO delay
-          error "bah!"
-      pid <- flip evalStateT 0 $ Immortal.create $ const computation
-      (do
-        threadDelay (5*10^5)
-        v <- atomically $ readTVar tv
-        v @?= 0)
-        `finally` Immortal.stop pid
-
```

Comment:
Why are the tests removed?

Reply:
They no longer compile - `StateT IO` is not an instance of `MonadUnliftIO`. That's part of the reason of switching to `unliftio` - weird stateful `monad-control` instances are ruled out.